v0/destinations/vero/transform.js

```diff
@@ -0,0 +1,142 @@
+const set = require("set-value");
+const get = require("get-value");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  getValueFromMessage,
+  isDefinedAndNotNull
+} = require("../../util");
+const { EventType } = require("../../../constants");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG, BASE_URL } = require("./config");
+
+// This function handles any tag add or removal requests.
+
+const tagResponseBuilder = async (message, tags, destination) => {
+  const response = defaultRequestConfig();
+
+  const payload = {};
+  const authToken = get(destination, "Config.authToken");
+  set(payload, "auth_token", authToken);
+
+  const id = getValueFromMessage(message, "userId");
+  set(payload, "id", id);
+
+  // add[] and remove[] arrays contain the tags to be added or removed.
+  const addTags = getValueFromMessage(tags, "add");
+  if (Array.isArray(addTags) && addTags.length > 0)
+    set(payload, "add", addTags);
+
+  const removeTags = getValueFromMessage(tags, "remove");
+  if (Array.isArray(removeTags) && removeTags.length > 0)
+    set(payload, "remove", removeTags);
+
+  response.endpoint = `${BASE_URL}/users/tags/edit`;
+  response.body.JSON = payload;
+  response.method = "PUT";
+
+  return response;
+};
+
+// This function handles the common response payload for the supported calls
+
+const commonResponseBuilder = async (message, category, destination) => {
+  const response = defaultRequestConfig();
+
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const authToken = get(destination, "Config.authToken");
+  set(payload, "auth_token", authToken);
+
+  const address = get(payload, "channels.address");
+  const platform = get(payload, "channels.platform");
+  /*
+    here we are adding a K-V pair in the channels object, i.e 
+    channels : {
+      "type" : "push"
+    }
+    This is a mandatory but also unchanging field for channels object.
+  */
+  if (isDefinedAndNotNull(address) && isDefinedAndNotNull(platform)) {
+    set(payload, "channels.type", "push");
+  }
```

Comment:
It looks like only an identify call parameter so no need to add it with other call payloads.

Reply:
Both address and platform variables will be undefined for all calls except for Identity call because payload.channels is defined only for Identity call. So the payload is not gonna have these parameters for calls other than Identity calls. 
Will add this info in the comments.